newIDE/app/src/ResourcesList/ResourcePreview/ImagePreview.js

```diff
@@ -147,6 +152,15 @@ const ImagePreview = (props: Props) => {
     }));
   };
 
+  const resourceIsSmooth = (
+    project: gdProject,
+    resourceName: string
+  ): boolean => {
+    const resource = project.getResourcesManager().getResource(resourceName);
+    const imageResource = gd.asImageResource(resource);
```

Comment:
Any cast is very risky: if the resource is not an image resource, you've instructed the C++/webAssembly code to consider something that is not an `ImageResource` to be handled as a `ImageResource`. This will result in memory corruption if you then try to modify something related to it.

Instead, use a check to verify the "type" of the class. Here you can use "getKind":
```
const resource = project.getResourcesManager().getResource(resourceName);
if (resource.getKind() !== 'image') return false; // The resource is not even an image.

const imageResource = gd.asImageResource(resource);
return imageResource.isSmooth();
```